### PR TITLE
Adding e2e-poc

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: server/e2e/sync_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStopPluginSendMessageAndReplyToMSTeamsLinkedChannelStartPlugin(t *testing.T) {
+	t.Skip("Not implemented yet")
```
Comment (wiggin77) [inline server/e2e/sync_test.go:18]: Is this still intended to be skipped?

File: server/e2e/main_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+var msClient msteams.Client
+var mmClient *model.Client4
+var mmClientAdmin *model.Client4
+var testCfg *TestConfig
+
+func setup(t *testing.T) {
+	if testCfg == nil {
+		data, err := os.ReadFile("testconfig.toml")
```
Comment (wiggin77) [inline server/e2e/main_test.go:43]: Nit:  would be nice if this error lead people to a small README that explained how to set this up.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Minor comments.

File: server/e2e/main_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+package main
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pelletier/go-toml/v2"
```
Comment (lieut-data) [inline server/e2e/main_test.go:13]: nit: almost all our configuration files are `json`, and I wonder if we should stick with that for consistency?

**Review by lieut-data — APPROVED** [review]

File: server/e2e/sync_test.go (hunk 1 above)
Comment (jespino) [reply server/e2e/sync_test.go:18]: Yes, this is for whenever we have the sync mechanism done

File: server/e2e/main_test.go (hunk 3 above)
Comment (jespino) [reply server/e2e/main_test.go]: Done, changed to json

**Pull request merged**